cli/mender-artifact/modify.go

```diff
@@ -43,25 +41,22 @@ func modifyArtifact(c *cli.Context) error {
 		return cli.NewExitError("File ["+c.Args().First()+"] does not exist.", 1)
 	}
 
-	pubKey, err := processModifyKey(c.String("key"))
-	if err != nil {
-		return cli.NewExitError("Error processing private key: "+err.Error(), 1)
-	}
-
 	modifyCandidates, isArtifact, err :=
-		getCandidatesForModify(c.Args().First(), pubKey)
+		getCandidatesForModify(c.Args().First())
 
 	if err != nil {
 		return cli.NewExitError("Error selecting images for modification: "+err.Error(), 1)
 	}
 	// strip the data and boot partitions
-	if isArtifact {
-		modifyCandidates = modifyCandidates[0:1]
-		for _, mc := range modifyCandidates {
-			defer os.Remove(mc.path)
+	if modifyCandidates != nil {
```

Comment:
Curious as to why this check is necessary, since the error is already handled above?

Reply:
I see now that my commit message is not as clear as I hoped for. I will try to clarify it here (this applies to other of your comments in the PR).

I have introduced a design change on this function: now it can return `nil` partitions on valid Artifacts that just happen to have no partitions (like Update Modules). Hence there is no error returned, but the caller still needs to check if there are partitions or not.

This is the detail I wanted to discuss in person.

Reply:
I we do this in order to support changing the name (and maybe in the future other attributes like device type for instance) on Artifacts that are not rootfs-image type

Reply:
I've had another look now. I mean, the PR doesnt interfere with the current code in any degrading way  -- which to be fair is good soup to begin with. However, maybe this would be a nice opportunity to make this more explicit. I feel like `getCandidatedForModify` might make this more explicit, maybe through returning some sort of an enum, instead of a bool, where `isArtifact` is currently at.
Thoughts?

Reply:
I thought about it at some point, actually. But now that a golang expert (you) suggests it, I will do it ;)

Reply:
haha, I'd be careful going down that road...